yt_dlp/extractor/nytimes.py

```diff
@@ -125,138 +91,427 @@ class NYTimesIE(NYTimesBaseIE):
+            'thumbnail': r're:https?://\w+\.nyt.com/images/.*\.jpg',
+            'duration': 1322,
         }
     }, {
-        'url': 'http://www.nytimes.com/2016/10/16/books/review/inside-the-new-york-times-book-review-the-rise-of-hitler.html',
+        'url': 'https://www.nytimes.com/2023/11/29/business/dealbook/kamala-harris-biden-voters.html',
+        'md5': '3eb5ddb1d6f86254fe4f233826778737',
         'info_dict': {
-            'id': '100000004709479',
-            'title': 'The Rise of Hitler',
-            'ext': 'mp3',
-            'description': 'md5:bce877fd9e3444990cb141875fab0028',
-            'creator': 'Pamela Paul',
-            'duration': 3475,
+            'id': '100000009202270',
+            'ext': 'mp4',
+            'title': 'Kamala Harris Defends Biden Policies, but Says ‘More Work’ Needed to Reach Voters',
+            'description': 'md5:de4212a7e19bb89e4fb14210ca915f1f',
+            'timestamp': 1701290997,
+            'upload_date': '20231129',
+            'uploader': 'By The New York Times',
+            'creator': 'Katie Rogers',
+            'thumbnail': r're:https?://\w+\.nyt.com/images/.*\.jpg',
+            'duration': 97631,
         },
         'params': {
+            # inconsistant md5
             'skip_download': True,
         },
     }, {
-        'url': 'http://www.nytimes.com/news/minute/2014/03/17/times-minute-whats-next-in-crimea/?_php=true&_type=blogs&_php=true&_type=blogs&_r=1',
+        # multiple videos in the same article
+        'url': 'https://www.nytimes.com/2023/12/02/business/air-traffic-controllers-safety.html',
+        'info_dict': {
+            'id': 'air-traffic-controllers-safety',
+            'title': 'Drunk and Asleep on the Job: Air Traffic Controllers Pushed to the Brink',
+            'description': 'md5:549e5a5e935bf7d048be53ba3d2c863d',
+        },
+        'playlist_count': 3,
+
+    }, {
+        'url': 'https://www.nytimes.com/2023/12/02/business/media/netflix-squid-game-challenge.html',
         'only_matching': True,
     }]
 
-    def _extract_podcast_from_json(self, json, page_id, webpage):
-        podcast_audio = self._parse_json(
-            json, page_id, transform_source=js_to_json)
+    def _extract_content_from_block(self, block):
+        details = traverse_obj(block, {
+            'id': ('sourceId', {str}),
+            'uploader': ('bylines', ..., 'renderedRepresentation', {str}),
+            'duration': (('duration', 'length', ...), {int_or_none}),
+            'upload_date': ('firstPublished', {unified_strdate}),
+            'timestamp': ('firstPublished', {parse_iso8601}),
+            'series': ('podcastSeries', {str})}, get_all=False)
+
+        media = traverse_obj(block, ('renditions', ...))
+        formats, subtitles = self._extract_media_from_json(details.get('id'), media)
+
+        # audio articles will have an url and no formats
+        url = traverse_obj(block, ('fileUrl'), get_all=False)
```

Comment:
`get_all=False` isn't needed here but a `url_or_none` transformation could help
```suggestion
        url = traverse_obj(block, ('fileUrl', {url_or_none}))
```